libraries/AP_Math/spline5.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * spline5.cpp
+ * Copyright (C) William Geyer and Chris Olson 2018
+ *
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdint.h>
+#include "spline5.h"
+
+void splinterp5(const float x[5], float out[5][4]){
+
+    // loop counter
+    uint8_t i;
+
+    // number of spline points
+    uint8_t n = 5;
+
+    // working variables
+    float p;
+    float u[5];
+
+    // second derivative
+    float z[5];
+
+    // set the second derivative to 0 at the ends
+    z[0] = u[0] = 0;
+    z[n-1] = 0;
+
+    // decomposition loop
+    for(i=1;i<n-1;i++){
+        p = 0.5f * z[i-1] + 2.0f;
+        z[i] = -0.5f / p;
+        u[i] = x[i+1] + x[i-1] - 2.0f * x[i];
+        u[i] = (3.0f * u[i] - 0.5f * u[i-1]) / p;
+    }
+
+    // back-substitution loop
+    for(i=n-1;i>0;i--){
+        z[i] = z[i] * z[i+1] + u[i];
+    }
+
+    for(i=0;i<n-1;i++){
```

Comment:
for(uint8_t i=0;i<n-1;i++) {

Reply:
why would I declare i in the for loop each time when I would just declare it once at the beginning of the function?  I could see declaring it in the for loop  if I only used i once.